vtex-apps/safedata
# Feature/perf metrics

#### What is the purpose of this pull request?
- Added memory caching on some objects (auth-info from VtexId, MD client entity and app settings)
- Added a prometheus custom metric "safedata_request_total" with "account", "entity" and "statusCode" labels
- Changed app scaling configuration

#### What problem is this solving?
This PR helps with the performance of the app, and allows creating useful dashboards in Grafana to monitor its usage.

#### How should this be manually tested?
You can test this URL on postman:
https://safe--sandboxbrdevbrl.myvtex.com/safedata/CL/search?_fields=id,firstName,lastName
It will return 401 when you're not logged in. It'll return 200 with your data if you're logged in to the store (You are logged in if you have a valid value for the `VtexIdclientAutCookie_sandboxbrdevbrl` Cookie, you can set it manually on postman)

You can check the metrics through [Grafana](http://grafana.ingress.vtex.io/explore?orgId=1&left=%5B%22now-15m%22,%22now%22,%22prometheus%20devs-a%22,%7B%22expr%22:%22sum%20by(entity,%20statusCode)%20(%20rate(safedata_request_total%5B5m%5D)%20)%22,%22requestId%22:%22Q-e6459b47-395f-43b6-8b6b-b67822f98898-0A%22,%22instant%22:false,%22range%22:true%7D%5D). It may take a few seconds to see the results.

#### Screenshots or example usage
<img width="1059" alt="Screen Shot 2021-06-18 at 10 27 36" src="https://user-images.githubusercontent.com/1629129/122568158-dae5f800-d01f-11eb-8a69-251b25549747.png">


#### Types of changes
